Use more smart pointers in WebSocketChannel

rwlbuis opened the pull request:
#### bdc9699ecaf0c0a700edd6c8ab30ba2b4aaaa531
<pre>
Use more smart pointers in WebSocketChannel
<a href="https://bugs.webkit.org/show_bug.cgi?id=292979">https://bugs.webkit.org/show_bug.cgi?id=292979</a>

Reviewed by Chris Dumez.

Use more smart pointers in WebSocketChannel.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect):
(WebKit::WebSocketChannel::logErrorMessage):

Canonical link: <a href="https://commits.webkit.org/295093@main">https://commits.webkit.org/295093@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d36e9948825cc1cc29a4c22d5a36e04db83b9f86

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104081 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23785 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14105 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109277 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54749 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24152 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32329 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79070 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107087 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18758 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/93885 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59397 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/18560 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54109 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/88281 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/11989 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111663 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31237 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23027 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88079 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31601 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90080 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87736 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22330 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32618 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10367 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25685 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31166 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/36478 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/30960 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34296 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32520 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
```diff
@@ -147,14 +148,15 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
     StoredCredentialsPolicy storedCredentialsPolicy { StoredCredentialsPolicy::Use };
-    if (auto* frame = m_document ? m_document->frame() : nullptr) {
-        RefPtr mainFrame = m_document->localMainFrame();
+    document = m_document.get();
```
Why is this here?

annevk replied:
@rwlbuis this still needs addressing. With that I think this is good to go, but might be good to let @youennf confirm.

rwlbuis replied:
I was initially worried that webSocketConnectRequest call may change m_document, maybe through IPC calls etc, but I checked it and it simply builds a specific request, so that line should not be needed, I removed it.

annevk commented on Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
```diff
@@ -147,14 +148,15 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
-        if (auto* mainFrameDocumentLoader = mainFrame->document() ? mainFrame->document()->loader() : nullptr) {
-            auto* policySourceDocumentLoader = mainFrameDocumentLoader;
+        if (RefPtr mainFrameDocumentLoader = mainFrame->document() ? mainFrame->document()->loader() : nullptr) {
+            RefPtr policySourceDocumentLoader = mainFrameDocumentLoader;
```
I don't really understand the use of this either. Also not in the old code.

rwlbuis replied:
With old code, do you mean the code as it is on trunk?

annevk replied:
Yeah, why are we creating a new variable. Can't we just use the one we already have?

rwlbuis replied:
I think it was done this way since policySourceDocumentLoader can be a mainFrameDocumentLoader but also other values. I think policySourceDocumentLoader is the more important name here though so I think we can just drop the  mainFrameDocumentLoader variable, like I did in my latest upload.

cdumez commented on Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
```diff
@@ -147,16 +148,15 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
     StoredCredentialsPolicy storedCredentialsPolicy { StoredCredentialsPolicy::Use };
-    if (auto* frame = m_document ? m_document->frame() : nullptr) {
-        RefPtr mainFrame = m_document->localMainFrame();
+    if (RefPtr frame = document ? document->frame() : nullptr) {
```
The null check here doesn't make sense since you've already done it at the beginning on this function and early returned when null. Previously, we were checking a WeakPtr so it could in theory become null after calling `client->didUpgradeURL()` above for example. However, this is no longer possible after your code change.

rwlbuis replied:
Removed the null check.

cdumez commented on Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
```diff
@@ -147,16 +148,15 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
         pageID = mainFrame->pageID();
-        if (auto* mainFrameDocumentLoader = mainFrame->document() ? mainFrame->document()->loader() : nullptr) {
-            auto* policySourceDocumentLoader = mainFrameDocumentLoader;
+        if (RefPtr policySourceDocumentLoader = mainFrame->document() ? mainFrame->protectedDocument()->loader() : nullptr) {
```
Would be nice to store `mainFrame->document()` in a local variable since we call `protectedDocument()` twice.

rwlbuis replied:
I think actually the second protectedDocument() call is done on another frame, so I left this part as-is.

cdumez reviewed (approved):
LGTM with nits.

rwlbuis committed:
Use more smart pointers in WebSocketChannel
https://bugs.webkit.org/show_bug.cgi?id=292979

Reviewed by Chris Dumez.

Use more smart pointers in WebSocketChannel.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect):
(WebKit::WebSocketChannel::logErrorMessage):

Canonical link: https://commits.webkit.org/295093@main
